source/game.programme.newsevent.bmx

```diff
@@ -749,7 +776,7 @@ Type TNewsEvent Extends TBroadcastMaterialSource {_exposeToLua="selected"}
 		'set happened time, add to collection list...
 		GetNewsEventCollection().setNewsHappened(Self, time)
```

Comment:
Ohne nochmal genauer auf die Stellen zu schauen, die doHappen aufrufen, sieht der Code etwas verdächtig aus...
* es wird explizit ein einem Event doHappen aufgerufen
* durch die EventCollection wird der Zeitpunkt in jedem Fall gesetzt
* die happen-Effekte werden aber nur ausgeführt, wenn time und GetWorldTime zusammenpassen

Wenn also doHappen mit einem "falschen" Zeitpunkt aufgerufen wird, bekommt das NewsEvent also einen Ausführungszeitpunkt, aber die Effekte würden nicht ausgelöst. Wird dann später für dasselbe Event nochmal doHappen aufgerufen? Beide möglichen Antworten wären seltsam.

Reply:
Den Code wollte ich jetzt nicht abaendern, da er nicht mit dem Fehler zu tun hat. Eine dortige Aenderung wird Folgen haben.

Reply:
Der Code wurde jetzt komplett ueberarbeitet.